### PR TITLE
Negotiate content type on post

### PR DESCRIPTION
**Pull request by mattgibson**
I ran into an error where the content type is not correctly passed through from the main Grape config on POST requests. This doesn't matter if using one of the default content types, but fails if it's a custom one.

This passes the options from the app to the formatter when it's called in the `JwtHandler`.

### CODE REVIEW

**Comment by mattgibson** [comment]
I don't understand the purpose of the failing specs. they seem to be asserting that a Grape API that has `format :json` set will provide a non-json response for errors. Is this right? It seems like that that is not something that the API in the specs should be doing.

File: spec/grape/jwt/grape_usage_spec.rb [hunk 1]
```diff
@@ -31,6 +31,49 @@ class API < Grape::API
...
+  end
+end
+
+module TestContentTypeWithoutVersioning
+  class API < Grape::API
+
+    # formatter :jsonapi, ->() {}
+    content_type :jsonapi, "application/vnd.api+json"
+    default_format :jsonapi
+    format :jsonapi
```
Comment (Jack12816) [inline spec/grape/jwt/grape_usage_spec.rb]: As I understand the Grape docs, you force the format with `format`. 

```
You can have your API only respond to a single format with format. If you use this, the API will not respond to file extensions other than specified in format. For example, consider the following API.
```

See: https://github.com/ruby-grape/grape#api-formats

---

Could you make another test case here were you allow xml and a custom jsonapi format for instance? (Without the forcing for a single one)

**Comment by Jack12816** [comment]
> I don't understand the purpose of the failing specs. they seem to be asserting that a Grape API that has `format :json` set will provide a non-json response for errors. Is this right? It seems like that that is not something that the API in the specs should be doing.

Mhh, I'm also a bit confused about this, too. Looks like I put a the wrong expectation at this point. Could you set it to the correct json strings? (I will meditate a bit on this)

File: lib/grape/jwt/authentication/jwt_handler.rb [hunk 2]
```diff
@@ -79,6 +79,12 @@ def parse_token(header)
           token
         end
 
+        def app_options
+          return @app.options if @app.instance_of?(Grape::Middleware::Formatter)
+          return @app.app.options if @app.respond_to?(:app) &&  @app.app&.instance_of?(Grape::Middleware::Formatter)
```
[the remark this answers is not in the document]
Comment (Jack12816) [reply lib/grape/jwt/authentication/jwt_handler.rb]: I don't get the point why `@app[.app].options` should be a `Grape::Middleware::Formatter`. Could you explain this a bit more? From my understanding it should be an options hash. 

Could you name this method then `formatter_options` and write a YARD doc block for it, like all other methods have?

**Comment by Jack12816** [comment]
@mattgibson Ping?

**Pull request closed without merging**